src/image-sequence-animator.tsx

```diff
@@ -82,30 +81,39 @@ const useUpdateState = (
       canvasRef.current.width = canvasRectBox.width * dpr;
       canvasRef.current.height = canvasRectBox.height * dpr;
 
-      state.current.containerWidth = canvasRectBox.width;
-      state.current.containerHeight = canvasRectBox.height;
+      setState(prevState => {
+        prevState.containerWidth = canvasRectBox.width;
+        prevState.containerHeight = canvasRectBox.height;
 
-      state.current.animationDistance =
-        stickyContainerBox.height -
-        props.framePaddingEnd -
-        props.framePaddingStart;
+        prevState.animationDistance =
+          stickyContainerBox.height -
+          props.framePaddingEnd -
+          props.framePaddingStart;
 
-      if (canvasRef.current) {
-        state.current.ctx = canvasRef.current.getContext(
-          '2d'
-        ) as CanvasRenderingContext2D;
-        state.current.ctx.scale(dpr, dpr);
-      }
+        if (canvasRef.current) {
+          prevState.ctx = canvasRef.current.getContext(
+            '2d'
+          ) as CanvasRenderingContext2D;
+          prevState.ctx.scale(dpr, dpr);
+        }
+
+        // 这里有些脏
+        // 假设要重构，那么应该考虑渲染所需内容不放在入参而放在作用域上，比如用useCallback
+        // 如果觉得useCallback会导致重渲判定那么可以配合useRef进行state缓存，参考：
+        // https://ahooks.js.org/hooks/advanced/use-persist-fn
+        drawImageSequence(props, canvasRef, prevState);
+
```

Comment:
是说，使用 
```suggestion
const drawImage = useCallback(() => {
  oldDrawImage(props, canvasRef, state)
}, [props, canvasRef, state])
```
这种形式？这样这个组件就和 React 的状态管理设计强绑定了，也会多不少的 diff 次数，我初衷是，这里的逻辑设计跟 React 是事实上无关的，同样的 draw 逻辑，搬到任何一个地方都适用~ 主要这也是从 @adobe/react-spectrum 学到的，他们也是通过这样的 `props`, `ref`, `stateRef` 入参的设计，快速的适配了 vue 和 svetle

Reply:
不是，是这种：
https://reactjs.org/docs/hooks-faq.html#how-to-read-an-often-changing-value-from-usecallback